### ISSUE TITLE
Emit a sync_complete Amplitude event

### ISSUE THREAD

**Issue by philbooth** [mozilla/fxa-amplitude-send#45]
The event `sync - sync_complete` is medium priority in the Amplitude taxonomy. I'm not sure how to emit it yet. Maybe it will be possible to work it out from the Sync data or maybe we could get the client to add it as an event.

(related discussion: https://github.com/mozilla/fxa-amplitude-send/issues/24#issuecomment-341173459)

**Comment by philbooth** [comment]
```
11:00:30 <pb> tcsc: so i've been poking around in s3 looking for stuff that will help me emit a sync_complete event to amplitude
11:01:13 <pb> tcsc: one of the data sets i've been looking at is called sync_flat_summary, which contains a field in the schema called engine_status
11:01:42 <pb> (and actually there's another one just called `status` too)
11:02:14 <pb> tcsc: any chance that might help me do what i want?
11:06:38 <pb> (in case it's helpful, the full schema i'm looking at looks like this...)
11:06:55 <pb> https://irccloud.mozilla.com/pastebin/yhE8Az7X/
11:52:39 ↔ ToBeFree and FlorinMezei|2 popped in  ↔ S, bogdan_maris and standups nipped out  
14:05:21 <tcsc> pb: unless you’re extremely clever, they’re irrelevant. They should only be present for non-success conditions
14:06:02 <tcsc> each row in sync_summary (e.g. not flattened) represents a sync though, does that help?
14:06:55 <tcsc> sync_flat_summary’s sync_id field also indicates distinct syncs
14:07:36 <pb> tcsc: oh, it proabably does. so just to be explicit, there is a 1:1 mapping between rows and successful syncs in sync_summary?
14:08:31 <tcsc> no, unsuccessful ones are there too. i meant that status/engine_status are for reporting problems
14:08:57 <tcsc> but yes, 1:1 for syncs and sync_summary rows
14:09:52 <pb> tcsc: got it. cool, thanks, sounds like i can scrabble it together from that
14:10:54 ⇐ s8321414 quit (s8321414@moz-cjcbva.dynamic-ip.hinet.net) Quit: Konversation terminated!
14:11:50 <tcsc> yeah. wrangling the timestamps sounds hard
14:12:02 <tcsc> but identifying syncs should be doable
```

**Comment by davismtl** [comment]
I would just like to raise a flag here. It occurred to me that this event may quadruple the number of events we send to Amplitude.

@philbooth take a look at this:
https://sql.telemetry.mozilla.org/queries/51687/source

Am I calculating this properly?

**Comment by philbooth** [comment]
> Am I calculating this properly?

Yep. (well, I'm using `status` rather than `failure_reason`, but the numbers end up about the same)

I'll hold off deploying #47 now, if the number of events is an issue. Fwiw, I did already import 1 day of data, so there's about 1.5 million of them in the prod Amplitude project already, sorry.

**Comment by davismtl** [comment]
I set up a call for us Monday morning just so we can go look it over together.

**Comment by philbooth** [comment]
We decided not to deploy this for now because of the event volume and Amplitude costs. Closing.

**Comment by davismtl** [comment]
@philbooth 
I think we had decided to hold off until we could get more value out of them (vs cert_signed event). This would likely entail cramming each event with a lot more properties.

Also, it would depend on whether @irrationalagent would get a lot of value from getting sync, sync errors and sync validation data in Amplitude. (vs Re:Dash)

Leif, what are your thoughts? Would Sync data in Amplitude really help you?

**Comment by irrationalagent** [comment]
I have an action item to define what properties we would theoretically send with a sync competed event. Will do that ASAP.

At top of my mind, having sync errors could **partly** allow us to analyze retention for those who experience them, and give us an indirect sense of what types of errors or failures are most noticed. For example, we think that syncing history is relatively invisible to the user (less likely to notice errors), while syncing bookmarks is more visible. If we could plot retention for users experiencing each type of error, we could generate some evidence for this hypothesis one way or another.

However I say "partly", because analyzing m-d retention based on syncs is difficult due to the fact that on Android syncing happens even when the app isn't actively used. Another thing on my plate is to define an event that would be more indicative of active use - if we had that, then the sync data could become more valuable with respect to analyzing retention, etc. 

As things stand now, we could probably analyze how often people actively uninstall after experiencing a sync error (because they drop out of the data completely), but nothing more subtle than that with regard to behavior. I'm less sure about this, bit we *might* also be able to also track whether triggering a repair of the bookmark tree moves the validation data in a positive way. 

Overall, we have sync error data in re:dash if we need to do basic performance monitoring currently. It's really on me to dig into this when I have more time in the coming week or so to understand if it will really add value to have this in amplitude. I might also know more when I finish my analysis of how syncing affects general browser performance. 

But let's hold off at least until I have a few concrete use-cases here. I'll update this when that happens.

**Comment by philbooth** [comment]
My bad! Re-opening and moving to backlog.

**Comment by irrationalagent** [comment]
maybe we should keep this one open as it might one day be useful for Ana
